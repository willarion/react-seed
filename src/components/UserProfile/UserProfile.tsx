import React from 'react';
import { Link } from 'react-router-dom';

interface UserProfileProps {
  to: string;
  alt: string;
}

export const UserProfile: React.FC<UserProfileProps> = ({ to, alt }) => {
  return (
    <Link to={to} className="profile">
      <span>User name</span>
      {/*TODO get username and image? from context of user*/}
      <img alt={alt} src="../../images/profilePhoto.png" />
    </Link>
  );
};
