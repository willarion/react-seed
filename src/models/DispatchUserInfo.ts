import React from 'react';

export interface DispatchUserInfo {
  dispatchUserInfo: React.Dispatch<{
    type: string;
    user: unknown;
    access: string;
  }>;
}
