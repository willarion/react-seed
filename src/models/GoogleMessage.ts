export interface GoogleMessage {
  id: string;
  snippet: string;
  payload: {
    headers: {
      name: string;
      value: string;
    }[];
    parts: {
      mimeType: string;
      body: {
        data?: string;
      };
      parts?: {
        mimeType: string;
        body: {
          data?: string;
        };
      }[];
    }[];
  };
}
