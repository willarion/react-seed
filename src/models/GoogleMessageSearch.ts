export interface GoogleMessageSearch {
  label?: 'in:inbox' | 'in:draft' | 'is:starred';
  category?: 'category:promotions' | 'category:social';
  searchInput?: string | string[];
}
