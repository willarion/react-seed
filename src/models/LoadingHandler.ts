export interface LoadingHandler {
  handleLoading: (flag: boolean) => void;
}
