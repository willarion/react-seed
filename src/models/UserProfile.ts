export interface UserProfile {
  googleId: string;
  imageUrl: string;
  email: string;
  name: string;
  givenName: string;
  familyName: string;
}
