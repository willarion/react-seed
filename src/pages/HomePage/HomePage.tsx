import React from 'react';
// import { Button } from '../../components/Button/Button';
import { Header } from '../../components/Header/Header';
import { Menu } from '../../components/Menu/Menu';
import { ContainerMain } from '../../components/ContainerMain/ContainerMain';
import { Submenu } from '../../components/Submenu/Submenu';

export const HomePage: React.FC = ({}) => {
  return (
    <section>
      <Header />
      <Menu />
      <ContainerMain>
        <Submenu />
      </ContainerMain>
    </section>
  );
};
