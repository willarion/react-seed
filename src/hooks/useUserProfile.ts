import React from 'react';
import { LoginContext } from '../App';
import { UserProfile } from '../models/UserProfile';

const useUserProfile = (): UserProfile | null => {
  return React.useContext(LoginContext).user;
};

export default useUserProfile;
