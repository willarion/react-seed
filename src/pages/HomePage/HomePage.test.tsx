import { shallow } from 'enzyme';
import { HomePage } from './HomePage';
import { UserProfile } from '../../models/UserProfile';
import useUserProfile from '../../hooks/useUserProfile/useUserProfile';
import { mocked } from 'ts-jest/utils';

jest.mock('../../hooks/useUserProfile/useUserProfile');

const userProfile: UserProfile = {
  googleId: 'string',
  imageUrl: 'string',
  email: 'string',
  name: 'string',
  givenName: 'string',
  familyName: 'string',
};

describe('<HomePage />', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should have render with user', () => {
    mocked(useUserProfile).mockImplementationOnce(() => userProfile);
    const wrapper = shallow(<HomePage />);
    expect(wrapper).toMatchSnapshot();
    expect(useUserProfile).toHaveBeenCalled();
  });

  it('should have render without user', () => {
    mocked(useUserProfile).mockImplementationOnce(() => null);
    const wrapper = shallow(<HomePage />);
    expect(wrapper).toMatchSnapshot();
    expect(useUserProfile).toHaveBeenCalled();
  });
});
