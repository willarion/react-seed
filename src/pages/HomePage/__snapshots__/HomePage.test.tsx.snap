// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePage /> should have render with user 1`] = `
<section>
  <Header />
  <Menu />
  <ContainerMain>
    <Submenu />
    <PostsContainer />
  </ContainerMain>
</section>
`;

exports[`<HomePage /> should have render without user 1`] = `
<section>
  <Header />
  <Menu />
  <ContainerMain>
    <Submenu />
  </ContainerMain>
</section>
`;
